srtcore/epoll.cpp

```diff
@@ -115,12 +115,11 @@ ENOMEM: There was insufficient memory to create the kernel object.
    if (++ m_iIDSeed >= 0x7FFFFFFF)
       m_iIDSeed = 0;
 
-   CEPollDesc desc;
-   desc.m_iID = m_iIDSeed;
-   desc.m_iLocalID = localid;
-   m_mPolls[desc.m_iID] = desc;
+   pair<map<int, CEPollDesc>::iterator, bool> res = m_mPolls.insert(pair<int, CEPollDesc>(m_iIDSeed, CEPollDesc(m_iIDSeed, localid)));
+   if (!res.second)  // Insertion failed (exists or no memory)
+      throw CUDTException(MJ_NOTSUP, MN_EIDINVAL);
```

Comment:
That's rather a wrong approach, as you have already allocated resources for locals. Besides, whether a key exists in the map, it can be as well checked in the beginning of this function, as executing none of the parts makes sense in this case. Also `EIDINVAL` isn't the error to be reported on internal resource errors, but in API functions where you supply an EID of a nonexistent CEPollDesc, while this function is only about to create it.